### PR TITLE
[Caffe2] Clean up ideep integration

### PR DESCRIPTION
**Pull request by yinghai**
Follow-up of https://github.com/pytorch/pytorch/pull/6699. 

Major fix is that if we don't have MKL-ML installed, current cmake config will still include its library and header which will lead to cmake error. 

Test Plan:
Build on my machine without `MKL-ML`
```
cmake .. -DUSE_CUDA=OFF -DUSE_MKL=ON -DBLAS=MKL -DMKL_USE_MKLML=OFF
```

Run:
```
(venv)[yinghai:python: (fix_ideep)]$ pytest -v ideep/
=========================================================================================================================================================================== test session starts ============================================================================================================================================================================
platform linux2 -- Python 2.7.5, pytest-3.4.0, py-1.5.2, pluggy-0.6.0 -- /data/users/yinghai/aml/venv/bin/python
cachedir: ../../.pytest_cache
rootdir: /data/users/yinghai/aml/pytorch, inifile:
plugins: cov-2.5.1, nbval-0.9.0, hypothesis-3.44.23
collected 16 items

ideep/LRN_op_test.py::LRNTest::test_LRN PASSED                                                                                                                                                                                                                                                                                                                       [  6%]
ideep/concat_split_op_test.py::TestConcatSplitOps::test_concat PASSED                                                                                                                                                                                                                                                                                                [ 12%]
ideep/concat_split_op_test.py::TestConcatSplitOps::test_split PASSED                                                                                                                                                                                                                                                                                                 [ 18%]
ideep/conv_op_test.py::ConvTest::test_convolution PASSED                                                                                                                                                                                                                                                                                                             [ 25%]
ideep/convfusion_op_test.py::ConvFusionTest::test_convolution_relu_fusion PASSED                                                                                                                                                                                                                                                                                     [ 31%]
ideep/convfusion_op_test.py::ConvFusionTest::test_convolution_sum_fusion PASSED                                                                                                                                                                                                                                                                                      [ 37%]
ideep/convfusion_op_test.py::ConvFusionTest::test_convolution_sum_relu_fusion PASSED                                                                                                                                                                                                                                                                                 [ 43%]
ideep/dropout_op_test.py::DropoutTest::test_dropout_is_test PASSED                                                                                                                                                                                                                                                                                                   [ 50%]
ideep/dropout_op_test.py::DropoutTest::test_dropout_ratio0 <- ../../../../../usr/lib64/python2.7/unittest/case.py SKIPPED                                                                                                                                                                                                                                            [ 56%]
ideep/elementwise_sum_op_test.py::ElementwiseSumTest::test_elementwise_sum PASSED                                                                                                                                                                                                                                                                                    [ 62%]
ideep/fc_op_test.py::FcTest::test_fc PASSED                                                                                                                                                                                                                                                                                                                          [ 68%]
ideep/pool_op_test.py::PoolTest::test_pooling PASSED                                                                                                                                                                                                                                                                                                                 [ 75%]
ideep/relu_op_test.py::ReluTest::test_relu PASSED                                                                                                                                                                                                                                                                                                                    [ 81%]
ideep/softmax_op_test.py::SoftmaxTest::test_softmax PASSED                                                                                                                                                                                                                                                                                                           [ 87%]
ideep/spatial_bn_op_test.py::TestSpatialBN::test_spatialbn_test_mode PASSED                                                                                                                                                                                                                                                                                          [ 93%]
ideep/spatial_bn_op_test.py::TestSpatialBN::test_spatialbn_train_mode PASSED                                                                                                                                                                                                                                                                                         [100%]

=================================================================================================================================================================== 15 passed, 1 skipped in 1.84 seconds ===================================================================================================================================================================
(venv)[yinghai:python: (fix_ideep)]$
```

cc: @gujinghui @jgong5

### CODE REVIEW

File: cmake/Modules/FindMKL.cmake [hunk 1]
```diff
@@ -17,6 +17,9 @@
 # ---[ Options
 include(CMakeDependentOption)
 
+option(MKL_USE_IDEEP "Use IDEEP interface in MKL BLAS" ON)
```
Comment (Maratyszcza) [inline cmake/Modules/FindMKL.cmake]: All options should be in top-level CMakeLists. BTW, any reason to not call it `USE_MKL_IDEEP`?

File: caffe2/python/ideep/convnet_benchmarks.py [hunk 2]
```diff
@@ -697,7 +697,7 @@ def GetArgumentParser():
         help="If True, dump the model prototxts to disk."
     )
     parser.add_argument("--net_type", type=str, default="simple")
-    parser.add_argument("--num_workers", type=int, default=1)
+    parser.add_argument("--num_workers", type=int, default=2)
```
Comment (Maratyszcza) [inline caffe2/python/ideep/convnet_benchmarks.py]: This change looks irrelevant to the rest

Comment (yinghai) [reply caffe2/python/ideep/convnet_benchmarks.py]: I was testing the stuff and set it to `1` and checked it into accidentally. Reverted it back to `2`.

**Review by Maratyszcza — APPROVED** [review]
Thanks, LGTM! Please rename MKL_USE_IDEEP to USE_IDEEP and MKL_USE_MKLML to USE_MKLML for consistency with other options.
